adam_core/propagator/propagator.py

```diff
@@ -105,30 +158,104 @@ def propagate_orbits(
             variants = None
 
         if max_processes is None or max_processes > 1:
-            with concurrent.futures.ProcessPoolExecutor(
-                max_workers=max_processes
-            ) as executor:
 
-                # Add orbits to propagate to futures
+            propagated_list: List[Orbits] = []
+            variants_list: List[VariantOrbits] = []
+
+            if parallel_backend == "cf":
+                with concurrent.futures.ProcessPoolExecutor(
+                    max_workers=max_processes
+                ) as executor:
+
+                    # Add orbits to propagate to futures
+                    futures = []
+                    for orbit_chunk in _iterate_chunks(orbits, chunk_size):
+                        futures.append(
+                            executor.submit(
+                                propagation_worker,
+                                orbit_chunk,
+                                times,
+                                self.__class__,
+                                **self.__dict__,
+                            )
+                        )
+
+                    # Add variants to propagate to futures
+                    if variants is not None:
+                        for variant_chunk in _iterate_chunks(variants, chunk_size):
+                            futures.append(
+                                executor.submit(
+                                    propagation_worker,
+                                    variant_chunk,
+                                    times,
+                                    self.__class__,
+                                    **self.__dict__,
+                                )
+                            )
+
+                    for future in concurrent.futures.as_completed(futures):
+                        result = future.result()
+                        if isinstance(result, Orbits):
+                            propagated_list.append(result)
+                        elif isinstance(result, VariantOrbits):
+                            variants_list.append(result)
+                        else:
+                            raise ValueError(
+                                f"Unexpected result type from propagation worker: {type(result)}"
+                            )
+
+            elif parallel_backend == "ray":
+
+                if RAY_INSTALLED is False:
+                    raise ImportError(
+                        "Ray must be installed to use the ray parallel backend"
+                    )
+
+                if not ray.is_initialized():
+                    ray.init(num_cpus=max_processes)
+
+                # Add orbits and times to object store
+                times_ref = ray.put(times)
```

Comment:
If there is any chance that orbits or times will need to be passed into future runs, the calling function (eg thor) will probably want to send in the refs rather than the literal values so that they don't get duplicated in memory.

I think it's likely we are going to have an unpleasant propagator signature here where we have this odd matrix of choices between parallel backend and propagator, worker functions, sending refs vs values, etc.

All that said, maybe it's easiest to allow a `orbits: Union[Orbits, ObjectRef]` here for now.

Reply:
I've explicitly added support for passing object references. Note that we do need to dereference orbits when we propagate covariances but otherwise the reference just gets passed along. 

Unit tests have also been added to test pass-by reference.